.github/workflows/ci.yml

```diff
@@ -14,17 +14,25 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, ubuntu-22.04, debian-10, debian-11, mariner-2]
-        arch: [arm, arm64, amd64]
-        exclude:
-          - os: ubuntu-22.04
-            arch: arm
-          - os: ubuntu-22.04
-            arch: arm64
-          - os: mariner-2
-            arch: arm
-          - os: mariner-2
-            arch: arm64
+        target:
+          [
+            { name: almalinux-9, arch: amd64 },
+            { name: amazonlinux-2, arch: amd64 },
+            { name: centos-7, arch: amd64 },
+            { name: centos-8, arch: amd64 },
+            { name: debian-10, arch: amd64 },
+            { name: debian-11, arch: amd64 },
+            { name: debian-12, arch: amd64 },
+            { name: mariner-2, arch: amd64 },
+            { name: oraclelinux-7, arch: amd64 },
+            { name: oraclelinux-8, arch: amd64 },
+            { name: rhel-7, arch: amd64 },
+            { name: rhel-8, arch: amd64 },
+            { name: rhel-9, arch: amd64 },
+            { name: sles-15, arch: amd64 },
+            { name: ubuntu-20.04, arch: amd64 },
+            { name: ubuntu-22.04, arch: amd64 },
```

Comment:
If all the targets are `amd64` we should be able to use container jobs directly which will result in the CI completing faster. 

https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container

Reply:
yeah good point, i rather not remove it for now as its likely we will re-introduce